src/nbl/video/CVulkanConnection.h

```diff
@@ -0,0 +1,79 @@
+#ifndef __NBL_C_VULKAN_CONNECTION_H_INCLUDED__
+#define __NBL_C_VULKAN_CONNECTION_H_INCLUDED__
+
+#include <volk.h>
+#include "nbl/video/IAPIConnection.h"
+#include "nbl/video/surface/CSurfaceVKWin32.h"
+#include "nbl/video/CVulkanPhysicalDevice.h"
+
+namespace nbl {
+namespace video
+{
+
+class CVulkanConnection final : public IAPIConnection
+{
+public:
+    CVulkanConnection(uint32_t appVer, const char* appName)
```

Comment:
I recommend `core::smart_refctd_ptr<CVulkanConnection> create` static method because if anything goes wrong in the constructor you're fucked (you have to throw an exception)